File: src/test/scala/net/igsoft/sdi/scala/testclasses/RCreator.scala
package net.igsoft.sdi.scala.testclasses

import net.igsoft.sdi.{Creator, InstanceCreator, ParameterBase}

case class RCreatorParams(name: String, surname: String) extends ParameterBase {
  override def uniqueId: String = name + surname
}

class RCreator extends Creator[R, RCreatorParams] {
  override def create(instanceCreator: InstanceCreator, rCreatorParams: RCreatorParams) = new R(instanceCreator.getOrCreate(classOf[Stepper]), rCreatorParams.name, rCreatorParams.surname)
}

File: src/test/java/net/igsoft/sdi/testclasses/RCreator.java
package net.igsoft.sdi.testclasses;

import net.igsoft.sdi.Creator;
import net.igsoft.sdi.InstanceCreator;

public class RCreator extends Creator<R, RCreatorParams> {
    @Override
    public R create(InstanceCreator instanceCreator, RCreatorParams rCreatorParams) {
        return new R(instanceCreator.getOrCreate(Stepper.class), rCreatorParams.getName(),
                     rCreatorParams.getSurname());
    }
}
